diff -up mplayer-export-2024-04-13/help/help_create_po.pl.omv~ mplayer-export-2024-04-13/help/help_create_po.pl
--- mplayer-export-2024-04-13/help/help_create_po.pl.omv~	2024-04-14 13:45:51.277584085 +0200
+++ mplayer-export-2024-04-13/help/help_create_po.pl	2024-04-14 13:48:07.163064410 +0200
@@ -57,6 +57,10 @@ open(po, "> $ARGV[1]") or die "Cannot op
 
 print po "# DO NOT EDIT - this file is generated from $ARGV[0]\n";
 
+print po "msgid \"\"\n";
+print po "msgstr \"\"\n";
+print po "\"Content-Type: text/plain; charset=UTF-8\"\n";
+
 foreach my $id (keys %podefs)
 {
   # for the moment only X11/GTK GUI strings
