--- MPlayer-1.3.0/configure.omv~	2018-01-11 12:48:28.199700453 +0100
+++ MPlayer-1.3.0/configure	2018-01-11 12:49:32.270875998 +0100
@@ -8187,7 +8187,7 @@ fi
 if test "$_dvdnav" = yes ; then
   def_dvdnav='#define CONFIG_DVDNAV 1'
   extra_cflags="$extra_cflags $($_dvdnavconfig --cflags)"
-  extra_ldflags="$extra_ldflags $($_dvdnavconfig --libs)"
+  extra_ldflags="$extra_ldflags $($_dvdnavconfig --libs) -ldvdread"
   inputmodules="dvdnav $inputmodules"
 else
   def_dvdnav='#undef CONFIG_DVDNAV'
